src/core/src/CoreMain.py

```diff
@@ -109,3 +119,16 @@ def __init__(self, argv):
 
             stdout_file_mirror.stop()
             file_logger.close(message_at_close="<End of output>")
+
+    @staticmethod
+    def update_patch_substatus_if_pending(patch_operation_requested, overall_patch_installation_operation_successful, patch_assessment_successful, status_handler, composite_logger):
+        if patch_operation_requested == Constants.INSTALLATION.lower() and not overall_patch_installation_operation_successful:
+            if not patch_assessment_successful:
+                status_handler.set_current_operation(Constants.INSTALLATION)
+                status_handler.add_error_to_status("Installation failed due to assessment failure. Please refer the error details in assessment substatus")
+            status_handler.set_installation_substatus_json(status=Constants.STATUS_ERROR)
+            composite_logger.log_debug('  -- Persisted failed installation substatus.')
+        if not patch_assessment_successful:
+            status_handler.set_assessment_substatus_json(status=Constants.STATUS_ERROR)
+            composite_logger.log_debug('  -- Persisted failed assessment substatus.')
```

Comment:
This would have been best encapsulated within the separate assessor and installer classes and invoked as needed. But that can't be done neatly as they may not have been initialized correctly. For completeness of positioning, it shouldn't be inside status handler as it's a level of detail that is not for status handler. It's not for CoreMain as it's usurping some logic that belongs elsewhere.

Maybe not for this PR, but consider what can be done for this.

Consider what may be done with this